citesphere/src/main/java/edu/asu/diging/citesphere/web/user/authorities/AuthorityEntryController.java

```diff
@@ -71,8 +136,21 @@
             logger.info("Not a valid URI: " + uri, e);
             return new ResponseEntity<>(HttpStatus.BAD_REQUEST);
         }
-     
-        entry = authorityService.create(entry, (IUser)authentication.getPrincipal());
+
+        entry = authorityService.create(entry, (IUser) authentication.getPrincipal());
+        return new ResponseEntity<IAuthorityEntry>(entry, HttpStatus.OK);
+    }
+    
+    @RequestMapping(value = "/auth/authority/add", method = RequestMethod.POST)
+    public ResponseEntity<IAuthorityEntry> addAuthorityEntry(Authentication authentication,
+            @RequestParam("uri") String uri) {
+        List<IAuthorityEntry> authorityEntryList = authorityConceptpowerResult.getFoundAuthorities().stream()
+                .filter(authority -> authority.getUri().equals(uri)).collect(Collectors.toList());
+        IAuthorityEntry entry = null;
+        if (authorityEntryList.size() > 0) {
+            entry = authorityService.create(authorityEntryList.get(0), (IUser) authentication.getPrincipal());
+        }
```

Comment:
this needs to take into account the source or it won't work once we add another authority service.